source/linux/Foundational_Components/U-Boot/UG-Memory.rst

```diff
@@ -1,973 +1,1120 @@
+         => fatload mmc 1 ${loadaddr} tiboot3.bin
+         => mmc write ${loadaddr} 0x0 0x800
+         => fatload mmc 1 ${loadaddr} tispl.bin
+         => mmc write ${loadaddr} 0x800 0x1000
+         => fatload mmc 1 ${loadaddr} u-boot.img
+         => mmc write ${loadaddr} 0x1800 0x2000
 
-.. _boot-linux-from-mmc:
+   .. ifconfig:: CONFIG_part_variant in ('J7200')
 
-Booting Linux from SD card or eMMC
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+      .. code-block:: console
+
+         => mmc dev 0 1
+         => fatload mmc 1 ${loadaddr} tiboot3.bin
+         => mmc write ${loadaddr} 0x0 0x800
+         => fatload mmc 1 ${loadaddr} tispl.bin
+         => mmc write ${loadaddr} 0x800 0x1000
+         => fatload mmc 1 ${loadaddr} u-boot.img
+         => mmc write ${loadaddr} 0x1800 0x2000
+
+   .. ifconfig:: CONFIG_part_variant in ('J721S2', 'AM62X', 'J784S4','J742S2', 'J722S')
+
+      .. code-block:: console
+
+         => mmc dev 0 1
+         => fatload mmc 1 ${loadaddr} tiboot3.bin
+         => mmc write ${loadaddr} 0x0 0x400
+         => fatload mmc 1 ${loadaddr} tispl.bin
+         => mmc write ${loadaddr} 0x400 0x1000
+         => fatload mmc 1 ${loadaddr} u-boot.img
+         => mmc write ${loadaddr} 0x1400 0x2000
+
+   .. ifconfig:: CONFIG_part_variant not in ('AM64X', 'J7200', 'J721S2', 'AM62X', 'J784S4','J742S2', 'J722S')
+
+      .. code-block:: console
+
+         => mmc dev 0 1
+         => fatload mmc 1 ${loadaddr} tiboot3.bin
+         => mmc write ${loadaddr} 0x0 0x400
+         => fatload mmc 1 ${loadaddr} tispl.bin
+         => mmc write ${loadaddr} 0x400 0x1000
+         => fatload mmc 1 ${loadaddr} u-boot.img
+         => mmc write ${loadaddr} 0x1400 0x2000
+         => fatload mmc 1 ${loadaddr} sysfw.itb
+         => mmc write ${loadaddr} 0x3600 0x800
+
+   eMMC layout
+   ^^^^^^^^^^^
+
+   .. ifconfig:: CONFIG_part_variant in ('AM64X')
+
+      .. code-block:: console
+
+                     boot0 partition (8 MB)                        user partition
+            0x0+----------------------------------+      0x0+-------------------------+
+               |     tiboot3.bin (1 MB)           |         |                         |
+         0x800+-----------------------------------+         |                         |
+               |       tispl.bin (2 MB)           |         |                         |
+         0x1800+----------------------------------+         |        rootfs           |
+               |       u-boot.img (4 MB)          |         |                         |
+         0x3800+----------------------------------+         |                         |
+               |      environment (128 KB)        |         |                         |
+         0x3900+----------------------------------+         |                         |
+               |   backup environment (128 KB)    |         |                         |
+         0x3A00+----------------------------------+         +-------------------------+
```

Comment:
So, I've seen sphinx parse these correctly in the past, but the spec actually says the first line of content in the code block dictates the indentation depth for the rest of the directive.

It may prevent future issues if you put the legend on the right side of the box and move the titles to the bottom of the code block.